pandas/core/indexes/api.py

```diff
@@ -206,6 +206,17 @@ def conv(i):
 
         return Index(lib.fast_unique_multiple_list([conv(i) for i in inds], sort=sort))
 
+    # GH 35092. Detect if we have an Index type for which the sort
+    # setting doesn't make sense
+    ind_types = [type(index) for index in indexes]
+    if any(
+        ind_type in [MultiIndex, RangeIndex, DatetimeIndex, CategoricalIndex]
```

Comment:
@gfyoung Turns out that `DatetimeIndex` breaks using eval to add a `Series` and a `DataFrame` in  `test_basic_series_frame_alignment` in `test_eval.py` but only on the [MacOS pipeline](https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=38693&view=logs&j=bef1c175-2c1b-51ae-044a-2437c76fc339&t=770e7bb1-09f5-5ebf-b63b-578d2906aac9&l=105). So I'm keeping this in the exception list too.

This is the last example of each of these types of indices breaking sorting expectations somewhere.